packages/vitest/src/node/core.ts

```diff
@@ -152,6 +152,9 @@ export class Vitest {
     await Promise.all(this._onSetServer.map(fn => fn()))
 
     this.projects = await this.resolveWorkspace(cliOptions)
+    const filteredProjects = toArray(resolved.project)
+    if (filteredProjects.length)
+      this.projects = this.projects.filter(p => filteredProjects.includes(p.getName()))
```

Comment:
Should this be configurable in watch-mode like other patterns are? Might be useful. 

https://github.com/vitest-dev/vitest/blob/d1e1bc90a9685ee153c17b12ab3d1c99d16d9750/packages/vitest/src/node/stdin.ts#L91-L114